### DISCUSSION TITLE
Different behaviour in version 5.2.x and 5.3.x related to datetime argument in celery task.

### DISCUSSION THREAD

**Discussion by sp1rs** [celery/celery#8283] [General]
In celery version **5.2.X**, if we pass the datetime object in the celery task as an argument, we receive a string in the task.

whereas in celery **5.3.X**, if we pass the datetime object in the celery task as an argument, we receive a datetime in the task.

```
@celery.task
def test_task(my_date):
    return f'in celery task the date is {my_date} and type is {type(my_date)}'


test_task.delay(my_date={'date': datetime.datetime.now()})
```

The output in celery 5.2.X is
```
'in celery task the date is {\'date\': \'2023-05-31T08:43:11.505263\'} and type is <class \'dict\'>'
```

The output in celery 5.3.X is
```
'in celery task the date is {\'date\': datetime.datetime(2023, 5, 31, 8, 45, 52, 285603)} and type is <class \'dict\'>'
```

Looks like a breaking change from 5.2 to 5.3

**Comment by sp1rs** [comment]
Okay, the behavior is because of this[ kombu change ](https://github.com/celery/kombu/pull/1515)

**Reply by auvipy** [reply]
the previous behavior was buggy. now it is better
